### PR TITLE
multipart: reject part upload if size is less than 5MB.

### PR DESCRIPTION
**Pull request by krishnasrinivas**

### CODE REVIEW

**Comment by harshavardhana** [comment]
This is a correct fix, but the tests would require some good changes. did you test this with S3 if this is the smallest multipart size?

**Comment by krishnasrinivas** [comment]
with S3, PutObjectPart succeeds for parts with < 5MB
But complete multipart failse with:

```
{ [S3Error: Your proposed upload is smaller than the minimum allowed size]
  code: 'EntityTooSmall',
  proposedsize: '1048576',
  minsizeallowed: '5242880',
  partnumber: '1',
  etag: 'b6d81b360a5672d80c27430f39153e2c',
  requestid: 'A5675AEAF4F07375',
  hostid: 'Ei6zxS8gWBHnK8TYsc/+TAlwftj9d6GFinPeh+4uMfEuHWgUOCJRil6B8ppjC65Yr37UB/d30yc=',
  amzRequestid: null,
  amzId2: null,
  amzBucketRegion: null } undefined

```

we can also do the same.

**Comment by krishnasrinivas** [comment]
@harshavardhana we are not able to send partnumber, proposedsize and minsizeallwoed in response XML as `APIErrorResponse` does not have those fields. We need to fix it sometime.

File: utils.go [hunk 1]
```diff
@@ -38,13 +38,20 @@ func checkValidMD5(md5 string) ([]byte, error) {
...
+	minPartSize = 1024 * 1024 * 5
 )
 
 // isMaxObjectSize - verify if max object size
 func isMaxObjectSize(size int64) bool {
 	return size > maxObjectSize
 }
 
+// Check if part size is less than allowed value.
+func isPartSizeLess(size int64) bool {
```
Comment (harshavardhana) [inline utils.go]: isMinPartSize() is better.

Comment (krishnasrinivas) [inline utils.go]: isMinPartSize - 'is minimum part size?" does not convey the meaning.

isPartSizeSmall or isPartSizeLess is better.

similarly `// isMaxObjectSize - verify if max object size` "verify if max object size" sentence is not right.

Comment (harshavardhana) [inline utils.go]: > isMinPartSize - 'is minimum part size?" does not convey the meaning.

isMinAllowedPartSize() would work, same for isMaxObjectSize() -> isMaxAllowedObjectSize().

**Comment by harshavardhana** [comment]
> @harshavardhana we are not able to send partnumber, proposedsize and minsizeallwoed in response XML as APIErrorResponse does not have those fields. We need to fix it sometime.

can you open a separate bug for this?

**Pull request merged**